### ISSUE TITLE
GitHub builds for (Tiny)Go fail 

### ISSUE THREAD

**Issue by tfenster** [fermyon/feedback#56]
**Describe the bug**
I used `spin gh create-action` to create my Github action and the first step worked smoothly

```
vscode ➜ /workspaces/verified-bluesky (main) $ spin gh create-action
Discovered Spin App: verified-bluesky at .
 - Go Component discovered at ./shared
 - Go Component discovered at ./validate-mvp
```

However, when I run that action, it fails because the step to install TinyGo fails

```
Run rajatjindal/setup-actions/tinygo@v0.0.1
  with:
    version: 0.33.0
  env:
    GO_VERSION: 1.[2](https://github.com/tfenster/verified-bluesky/actions/runs/11985785934/job/33418118471#step:4:2)3.2
    TINYGO_VERSION: 0.[3](https://github.com/tfenster/verified-bluesky/actions/runs/11985785934/job/33418118471#step:4:3)3.0
    SPIN_VERSION: 
setting up tinygo 0.33.0 on linux - amd6[4](https://github.com/tfenster/verified-bluesky/actions/runs/11985785934/job/33418118471#step:4:4)
Downloading tool from https://github.com/tinygo-org/tinygo/releases/download/0.33.0/tinygo0.33.0.linux-amd64.tar.gz
Creating tempdir /tmp/tmp/runner/40348f8b-bb9[5](https://github.com/tfenster/verified-bluesky/actions/runs/11985785934/job/33418118471#step:4:5)-47a3-9206-e99ecc4a[6](https://github.com/tfenster/verified-bluesky/actions/runs/11985785934/job/33418118471#step:4:6)14a
Error: Unexpected HTTP response: 404
```

Looking at the URL where it tries to download the URL, it seems like it is missing a `v`: It uses 
`https://github.com/tinygo-org/tinygo/releases/download/0.33.0/tinygo0.33.0.linux-amd64.tar.gz`
but it should use
`https://github.com/tinygo-org/tinygo/releases/download/v0.33.0/tinygo0.33.0.linux-amd64.tar.gz` 

You can see the issue here https://github.com/tfenster/verified-bluesky/actions/runs/11985785934/job/33418118471

**Frequency of the bug**
Please check one:
- [X] Consistent repro
- [ ] Intermittent repro
- [ ] No repro

**Set-up information**
`spin 3.0.0 (737778e 2024-11-07)`, error occurs during GitHub action run

**To Reproduce**
1. Use `spin gh create-action` for a Go-based spin application
2. Let the action run

**Expected behavior**
The action should complete successfully

**Screenshots**
![image](https://github.com/user-attachments/assets/30a2ae0e-f9ac-4c1a-8c08-2a085f0a2038)

**Additional context**
I couldn't find the place (probably a GitHub repo?) where the TinyGo install step is maintained, otherwise I would have created the report there. As it directly influences your builds, I still wanted to let you know and assume you have a contact

**Comment by ThorstenHans** [comment]
Thanks for reporting this @tfenster 

I was able to repo it and a fix is on its way. 

Until the new release of the `gh` plugin will be available through `spin plugins`, you can explicitly overwrite the tiny go version using `spin gh create-action --tinygo-version=v0.33.0 --overwrite`.

The `--overwrite` is only necessary if the GH Action workflow file (default path `.github/workflows/ci.yaml` already exists

**Comment by ThorstenHans** [comment]
The fix is now available on canary. You can install canary using the following commands

```bash
spin plugins uninstall gh
spin plugins install --url https://github.com/fermyon/spin-gh-plugin/releases/download/canary/gh.json
```

Once the referenced PR is merged, you can update the plugin using 

```bash
spin plugins update
spin plugins upgrade gh
```

**Comment by ThorstenHans** [comment]
Closing this, as we shipped `0.0.6` of the `gh` plugin now.

Thanks once again @tfenster
